src/lib/button-area/button-area.scss

```diff
@@ -1,11 +1,62 @@
-@use './mixins';
+@use './core' as *;
+@use '../state-layer';
+@use '../focus-indicator';
+@use '../utils/mixins' as utils;
 
-@include mixins.core-styles;
+//
+// Host
+//
+
+$host-tokens: [disabled-cursor focus-indicator-color];
 
 :host {
-  @include mixins.host;
+  @include tokens($includes: $host-tokens);
+}
+
+:host {
+  @include host;
 }
 
 :host([hidden]) {
   display: none;
 }
+
+//
+// Base
+//
+
+.forge-button-area {
+  @include tokens($excludes: $host-tokens);
+}
+
+.forge-button-area {
+  @include base;
+
+  &:not(.forge-button-area--disabled) {
+    &.forge-button-area {
+      @include enabled;
+    }
+  }
+
+  &.forge-button-area--disabled {
+    @include disabled;
+  }
+
+  &__button {
+    @include utils.visually-hidden;
+  }
```

Comment:
To build on my previous comments, we can probably simplify and remove nesting for these selectors once you remove BEM and use host attributes for things like `disabled`.